app/views/handlers/list_collector.py

```diff
@@ -55,5 +77,14 @@ def handle_post(self):
             # wtforms Form parents are not discoverable in the 2.3.3 implementation
             self.questionnaire_store_updater.update_answers(self.form.data)
             self.questionnaire_store_updater.save()
-        else:
+        elif self._is_list_collector_complete():
             return super().handle_post()
+
+    def _is_list_collector_complete(self):
+        if not self.repeating_block_ids:
+            return True
+
+        list_model = self._questionnaire_store.list_store.get(self.list_name)
+        return all(
+            self.questionnaire_store_updater.is_section_complete(section_id=self.current_location.section_id, list_item_id=list_item_id)
+            for list_item_id in list_model.items)
```

Comment:
Can this not use `get_first_incomplete_repeating_block_location`?

Reply:
Another nice catch - the method will essentially just be calling that method and returning the inverted bool interpretation of the result so arguably could be removed, but I still think it's worth keeping so the usage of `get_first_incomplete_repeating_block_location` for the purpose of list collector completeness is clear.

Reply:
I think the usage of `get_first_incomplete_repeating_block_location` doesn't become unclear if we did this?.
I think it is beneficial because:
1. Avoid having another bit of logic to get the same end result
2. Keeps it consistent with how we already do it. This is what happen in the `is_path_complete` method, which inverts `get_first_incomplete_...` method.
3. Personally, I don't think it is unclear either, currently it is, all sections for the list item are complete, it would just become, all sections don't have an incomplete or not any section have incomplete blocks.

If I had to choose one reason it would simply be for consistency. See what others think and go with consensus.

Reply:
Sorry I wasn't meaning that I was keeping this method as is, I agree with you that it should be changed to use the inverted result of `get_first_incomplete_repeating_block_location`, which has now been implemented

Reply:
Ah cool, I probably misunderstood. 👍

Reply:
Bad phrasing from me, is my bad